junit-jupiter-migration-support/src/main/java/org/junit/jupiter/migrationsupport/rules/ParameterizedExtension.java

```diff
@@ -0,0 +1,292 @@
+		}
+
+		return parameters[parameterIndex];
+	}
+
+	/**
+	 * Retrieves the Object[] of the current iteration we are working on.
+	 *
+	 * @param extensionContext the extensionContext
+	 * @param parameterCount the amount of parameters of the constructor.
+	 *
+	 * @return the Object[] for this parameter iteration.
+	 * @throws ParameterResolutionException If the amount of arguments of the constructor doesn't match the amount
+	 * of arguments of the currently resolved object[]
+	 */
+	private Object[] resolveParametersForConstructor(ExtensionContext extensionContext, int parameterCount)
+			throws ParameterResolutionException {
+		// @formatter:off
+		return parameters(extensionContext).map(ArrayList::new)
+				.map(l -> l.get(this.parametersCollectionIndex))
+				.filter(params -> params.length == parameterCount)
+				.orElseThrow(ParameterizedExtension::unMatchedAmountOfParametersException);
+		// @formatter:on
+	}
+
+	private static boolean hasCorrectParameterFields(ExtensionContext context) {
+		List<Field> fields = parametersFields(context);
+		boolean hasFieldInjection = !fields.isEmpty();
+
+		if (hasArgsConstructor(context) && hasFieldInjection) {
+			return false;
+		}
+		else if (hasFieldInjection) {
+			return areParametersFormedCorrectly(fields);
+		}
+
+		return true;
+	}
+
+	private static boolean areParametersFormedCorrectly(List<Field> fields) {
+		List<Integer> parameterValues = parameterIndexes(fields);
+
+		List<Integer> duplicateIndexes = duplicatedIndexes(parameterValues);
+
+		boolean hasAllIndexes = indexRangeComplete(parameterValues);
+
+		return hasAllIndexes && duplicateIndexes.isEmpty();
+	}
+
+	private static List<Integer> parameterIndexes(List<Field> fields) {
+		// @formatter:off
+		return fields.stream()
+				.map(f -> f.getAnnotation(Parameterized.Parameter.class))
+				.map(Parameterized.Parameter::value)
+				.collect(toList());
+		// @formatter:on
+	}
+
+	private static List<Integer> duplicatedIndexes(List<Integer> parameterValues) {
+		// @formatter:off
+		return parameterValues.stream().collect(groupingBy(identity())).entrySet().stream()
+				.filter(e -> e.getValue().size() > 1)
+				.map(Map.Entry::getKey)
+				.collect(toList());
+		// @formatter:on
+	}
+
+	private static Boolean indexRangeComplete(List<Integer> parameterValues) {
+		// @formatter:off
+		return parameterValues.stream()
+				.max(Integer::compareTo)
+				.map(i -> parameterValues.containsAll(IntStream.range(0, i).boxed().collect(toList())))
+				.orElse(false);
+		// @formatter:on
+	}
+
+	private static boolean lastParameterToBeResolved(ParameterContext parameterContext) {
+		return parameterContext.getIndex() == parameterContext.getDeclaringExecutable().getParameterCount() - 1;
+	}
+
+	private static Optional<Collection<Object[]>> parameters(ExtensionContext context) {
+		return context.getStore(parameters).getOrComputeIfAbsent("parameterMethod",
+			k -> new ParameterWrapper(callParameters(context)), ParameterWrapper.class).getValue();
```

Comment:
@marcphilipp Yes, I already did here https://github.com/junit-team/junit5/pull/723/files#diff-c8fe8d61d6bba040fc43c22d9db3f16aR291 (Looking at it now it could definitely have a better name) 

So in the current implementation the parameters method is indeed only called once, but it does still require storing them in the store. My comment was intended to show that even though the extensions are now local, I still need to store them for this behaviour.